A-1.c

```diff
@@ -0,0 +1,196 @@
+
+
+//! Reads [dest_size] number of integers from [stream] and writes to [dest] array.
+//! Returns true is specified number of elements was successfully written. False otherwise.
+bool get_int_array(FILE *stream, size_t dest_size, int *dest);
+//! Finds maximum value of [src_size] elements of [src] array.
+//! Returns found value or 0 if passed [src] pointer is invalid.
+int find_max(int *src, size_t src_size);
+//! Finds maximum value of [src_size] elements of [src] array that is less than [limit].
+//! Returns found value or 0 if passed [src] pointer is invalid.
+int find_limited_max(int *src, size_t src_size, int limit);
+//! Finds [dest_count] number of maximum values of [src_size] elements of [src] array.
+//! Returns address of newly allocated memory if passed [src] pointer is valid. NULL otherwise.
+int* find_n_max(int *src, size_t src_size, size_t dest_count);
+
+
+
+int main()
+{
+    size_t array_size = 0;
+    if(scanf("%lu", &array_size) != 1)
+    {
+        LOG_ERROR;
+        return 0;
+    }
+
+    int *array = (int*)malloc(array_size * sizeof(int));
+    if(!array)
+    {
+        LOG_ERROR;
+        return 0;
+    }
+    if(!get_int_array(stdin, array_size, array))
+    {
+        LOG_ERROR;
+        free(array);
+        return 0;
+    }
+
+    size_t max_count = 0;
+    if((scanf("%lu", &max_count) != 1) || (max_count > array_size))
+    {
+        LOG_ERROR;
+        free(array);
+        return 0;
+    }
+
+    int *res = find_n_max(array, array_size, max_count);
+    if(!res)
+    {
+        LOG_ERROR;
+        free(array);
+        return 0;
+    }
+
+    LOG_ARR_INT(res, max_count);
+
+    free(array);
+    free(res);
+
+    return 0;
+}
+
+
+
+bool get_int_array(FILE *stream, size_t dest_size, int *dest)
+{
+    if(!stream || !dest)
+    {
+        return false;
+    }
+
+    for(size_t i = 0; i < dest_size; ++i)
+    {
+        if(feof(stream) || (fscanf(stream, "%d", &dest[i]) != 1))
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+
+int find_max(int *src, size_t src_size)
+{
+    if(!src)
+    {
+        return 0;
+    }
+
+    int result = src[0];
+    for(size_t i = 0; i < src_size; ++i)
+    {
+        if(src[i] > result)
+        {
+            result = src[i];
+        }
+    }
+
+    return result;
+}
+
+
+
+int find_limited_max(int *src, size_t src_size, int limit)
+{
+    if(!src)
+    {
+        return 0;
+    }
+
+    int result = src[0];
+    for(size_t i = 0; i < src_size; ++i)
+    {
+        if((result >= limit) || (src[i] > result && src[i] < limit))
+        {
+            result = src[i];
+        }
+    }
+
+    return result;
+}
+
+
+
+int* find_n_max(int *src, size_t src_size, size_t dest_count)
+{
+    if(!src)
+    {
+        return NULL;
+    }
+
+    if(dest_count > src_size)
+    {
+        dest_count = src_size;
+    }
+
+    int *result = (int*)malloc(dest_count * sizeof(int));
+    if(!result)
+    {
+        return NULL;
+    }
+
+    if(dest_count < 1)
```

Comment:
dest_count - это size_t, значения < 1, это только 0. Зачем нужно возвращать result != 0 для массива 0 размера?

Reply:
Изменил
[commit](https://github.com/tp-deva/hw-babayan/pull/1/commits/08d922d2f03de2b682830ed69e8d63197564ac34)
Теперь в случае, если dest_count == 0, функция возвращает NULL.

```suggestion
if(max_count == 0)
{
    free(array);
    return 0;
}
```
Добавил для соответсвия тестам. В случае, если желаемое кол-во максимальных элементов равно нулю, программа завершается без выведения сообщения об ошибке.
```suggestion
--- Input ---
5
42 13 100 121 1
0
--- Output ---

```